deps/chakrashim/src/v8persistent.cc

```diff
@@ -61,14 +61,14 @@ template <class Callback, class Func>
 void SetObjectWeakReferenceCallbackCommon(
     JsValueRef object,
     Callback callback,
-    std::shared_ptr<WeakReferenceCallbackWrapper>* weakWrapper,
+    STD_SHARED_PTR<WeakReferenceCallbackWrapper>* weakWrapper,
     const Func& initWrapper) {
   if (callback == nullptr || object == JS_INVALID_REFERENCE) {
     return;
   }
 
   if (!*weakWrapper) {
-    *weakWrapper = std::make_shared<WeakReferenceCallbackWrapper>();
+    (*weakWrapper).reset(new WeakReferenceCallbackWrapper());
```

Comment:
May just want to avoid using shared_ptr and implement manual cleanup, and avoid the dependency and turning on RTTI,